core/tests/network_dial_error.rs

```diff
@@ -83,6 +84,70 @@ fn deny_incoming_connec() {
     .unwrap();
 }
 
+#[test]
+fn invalid_peer_id() {
+    // Checks whether dialling with the wrong peer id raises the correct error
+
+    let mut swarm1 = test_network(NetworkConfig::default());
+    let mut swarm2 = test_network(NetworkConfig::default());
+
+    swarm1.listen_on("/memory/0".parse().unwrap()).unwrap();
+
+    let address = async_std::task::block_on(future::poll_fn(|cx| match swarm1.poll(cx) {
```

Comment:
```suggestion
    let address = futures::executor::block_on(future::poll_fn(|cx| match swarm1.poll(cx) {
```

Just a nit pick. We will always depend on `futures` but might now always depend on `async_std`.

Reply:
okay, will fix all occurrences in that file